python/python/lance/dataset.py

```diff
@@ -1072,13 +1072,19 @@ def update(
         where : str, optional
             A SQL predicate indicating which rows should be updated.
 
+        Returns
+        -------
+        updates : dict
+            A dictionary containing the number of rows updated.
```

Comment:
```suggestion
        statistics : dict
            A dictionary containing statistics about the update operation
```
Minor nit: the word `updates` feels as if it is returning the updated values themselves.